controller/backup_controller.go

```diff
@@ -142,10 +150,23 @@ func (bc *BackupController) handleErr(err error, key interface{}) {
 		return
 	}
 
-	if bc.queue.NumRequeues(key) < maxRetries {
-		bc.logger.WithError(err).Warnf("Error syncing Longhorn backup %v", key)
-		bc.queue.AddRateLimited(key)
-		return
+	// The resync period of the backingimage is one hour and the maxRetries is 3.
+	// Thus, the deletion failure of the backup in error state is caused by the shutdown of the replica during backing up,
+	// if the lock hold by the backup job is not released.
+	// The workaround is to increase the maxRetries number and to retry the deletion until the lock acquisition
+	// of the backup is timeout after 150 seconds.
+	if strings.Contains(err.Error(), "failed lock") {
+		if bc.queue.NumRequeues(key) < maxRetriesOnAcquireLockError {
+			bc.logger.WithError(err).Warnf("Error syncing Longhorn backup %v", key)
```

Comment:
Do you think we should be able to differentiate the log message of line 160 and line 166?

Reply:
Better. I've updated the error messages.